### PR TITLE
Add BUILD_WITHOUT_DEPENDENCIES CMake option and nodeps CMake preset

### PR DESCRIPTION
**Pull request by traversaro**
This PR adds the `BUILD_WITHOUT_DEPENDENCIES` CMake option (that by default is `OFF`) and the `nodeps` CMake preset to easily enable `BUILD_WITHOUT_DEPENDENCIES` and `INSTALL_ALL_ROBOTS` CMake option.

The `BUILD_WITHOUT_DEPENDENCIES` CMake options drops any dependency on YCM, YARP and ICUBcontrib with the following limitations:
* The files are only installed, they are not copied in the build folder (as `yarp_install` does instead)
* There is no `uninstall` target (but unix users can easily uninstall anyhow with `xargs rm < install_manifest.txt`)
* There is no automatic setting of `CMAKE_INSTALL_PREFIX` to the ICUBcontrib install prefix

Furthermore, a `nodeps` CMake preset is added, to permit to easily build and install the CMake project with the options `BUILD_WITHOUT_DEPENDENCIES` and `INSTALL_ALL_ROBOTS` option enabled.

In particular, this functionality can be used to easily create an install of robots-configuration with the commands and make it available to YARP with:

~~~
git clone https://github.com/robotology/robots-configuration
cd robots-configuration
cmake --preset nodeps
cmake --install ./build
export YARP_DATA_DIRS=$YARP_DATA_DIRS:$(pwd)/build/install/share/ICUBcontrib
~~~

As this workflow is quite advanced and is meant just to be used in specific IIT-internal setups, I would avoid to document it in the README, and just document it where it is going to be used.

### CODE REVIEW

**Comment by traversaro** [comment]
This functionality is meant to be used with the option added to robotology-superbuild in https://github.com/robotology/robotology-superbuild/pull/1775 .

File: CMakeLists.txt [hunk 1]
```diff
@@ -111,4 +131,6 @@ else()
     endif()
 endif()
 
-icubcontrib_add_uninstall_target()
+if(NOT BUILD_WITHOUT_DEPENDENCIES)
+    icubcontrib_add_uninstall_target()
+endif()
```
Comment (S-Dafarra) [inline CMakeLists.txt:136]:
```suggestion
endif()

```

**Comment by S-Dafarra** [comment]
> git clone https://github.com/robotology/robots-configuration
> cd robots-configuration
> cmake --preset nodeps
> cmake --install ./build
> export YARP_DATA_DIRS=$YARP_DATA_DIRS:$(pwd)/build/install/share/ICUBcontrib

Just to understand, how is the install prefix set?

**Comment by traversaro** [comment]
> Just to understand, how is the install prefix set?

That is part of the CMake preset, i.e. in https://github.com/robotology/robots-configuration/pull/697/files#diff-1712b235e02bd4a3ad866fe26447a6dc5c1b449faeea513f449d244d46d03cf5R21 . Note that that is only a default, so you can still change it by calling `cmake --preset nodeps -DCMAKE_INSTALL_PREFIX=/another/install/prefix`.

**Comment by S-Dafarra** [comment]
> > git clone [robotology/robots-configuration](https://github.com/robotology/robots-configuration?rgh-link-date=2024-12-17T15%3A32%3A20Z)
> > cd robots-configuration
> > cmake --preset nodeps
> > cmake --install ./build
> > export YARP_DATA_DIRS=$YARP_DATA_DIRS:$(pwd)/build/install/share/ICUBcontrib
> 
> Just to understand, how is the install prefix set?

Saw it now: https://github.com/traversaro/robots-configuration/blob/5b267a80e38733695a8398c3db342a17b01dce66/CMakePresets.json#L21

I thought it was set somewhere in the ``CMakeLists.txt``

**Review by S-Dafarra — APPROVED** [review]

**Comment by traversaro** [comment]
> > > git clone [robotology/robots-configuration](https://github.com/robotology/robots-configuration?rgh-link-date=2024-12-17T15%3A32%3A20Z)
> > > cd robots-configuration
> > > cmake --preset nodeps
> > > cmake --install ./build
> > > export YARP_DATA_DIRS=$YARP_DATA_DIRS:$(pwd)/build/install/share/ICUBcontrib
> > 
> > 
> > Just to understand, how is the install prefix set?
> 
> Saw it now: https://github.com/traversaro/robots-configuration/blob/5b267a80e38733695a8398c3db342a17b01dce66/CMakePresets.json#L21
> 
> I thought it was set somewhere in the `CMakeLists.txt`

Yes, it may break the [POLA](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) if you are not used to `CMakePresets.json`, but now that we can start to drop Ubuntu 20.04 support, I think we can start using them as they are quite a powerful tool that can help declutter the CMakeLists.txt.

**Review by pattacini — APPROVED** [review]
Thanks @traversaro for your contribution!

**Pull request merged**